src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/Common/AsyncDebugScope.cs

```diff
@@ -101,6 +101,8 @@ public void Dispose()
                     _command.CompletePendingReadWithSuccess(true);
                 }
             }
+            _reader.Dispose();
```

Comment:
We need null check before calling Dispose(). Same to _command

Reply:
Moved them in blocks above